cmd/testscript/testdata/env_var.txt

```diff
@@ -0,0 +1,16 @@
+# Test passing environment variables to scripts
```

Comment:
Just to sanity check, how about adding a .gomodproxy entry to file.txt and check that the env var entries from that are correctly overridden (e.g. GOARCH, GOPATH)?

Reply:
Per our chat, completely agree. 

I've actually added a more comprehensive set of tests around this. Two files, one with `go` in the `PATH`, the other without. 

The matrix of scenarios tested is then:

* with and without `go` in `PATH
* with `.gomodproxy` and without
* with `GOPROXY` set in env and without
* with `-e` set and without

This also helps to confirm that setting a custom `GOPROXY` atop a `.gomodproxy` being present in the script results in the former, i.e. the override behaviour as described. 

The code before was just wrong; the vars were applied first when (obviously) they should have been applied last.